code/modules/projectiles/guns/ballistic/automatic.dm

```diff
@@ -235,116 +204,77 @@
 	slot_flags = 0
 	mag_type = /obj/item/ammo_box/magazine/mm712x82
 	weapon_weight = WEAPON_HEAVY
-	fire_sound = 'sound/weapons/gunshot_smg.ogg'
 	var/cover_open = FALSE
 	can_suppress = FALSE
 	burst_size = 3
 	fire_delay = 1
 	spread = 7
 	pin = /obj/item/firing_pin/implant/pindicate
+	bolt_type = BOLT_TYPE_OPEN
+	mag_display = TRUE
+	mag_display_ammo = TRUE
+	fire_sound = 'sound/weapons/rifleshot.ogg'
+	rack_sound = 'sound/weapons/chunkyrack.ogg'
 
 /obj/item/gun/ballistic/automatic/l6_saw/unrestricted
 	pin = /obj/item/firing_pin
 
 
 /obj/item/gun/ballistic/automatic/l6_saw/examine(mob/user)
 	..()
+	to_chat(user, "<b>ctrl + click</b> to [cover_open ? "close" : "open"] the dust cover.")
 	if(cover_open && magazine)
 		to_chat(user, "<span class='notice'>It seems like you could use an <b>empty hand</b> to remove the magazine.</span>")
 
 
-/obj/item/gun/ballistic/automatic/l6_saw/attack_self(mob/user)
+/obj/item/gun/ballistic/automatic/l6_saw/AltClick(mob/user)
 	cover_open = !cover_open
 	to_chat(user, "<span class='notice'>You [cover_open ? "open" : "close"] [src]'s cover.</span>")
 	if(cover_open)
 		playsound(user, 'sound/weapons/sawopen.ogg', 60, 1)
 	else
-		playsound(user, 'sound/weapons/sawclose.ogg', 60, 1)
+		playsound(user, 'sound/weapons/sawopen.ogg', 60, 1)
 	update_icon()
 
 
-/obj/item/gun/ballistic/automatic/l6_saw/update_icon()
-	icon_state = "l6[cover_open ? "open" : "closed"][magazine ? CEILING(get_ammo(0)/12.5, 1)*25 : "-empty"][suppressed ? "-suppressed" : ""]"
-	item_state = "l6[cover_open ? "openmag" : "closedmag"]"
+/obj/item/gun/ballistic/automatic/l6_saw/update_icon() 
+	. = ..()
+	add_overlay("l6_door_[cover_open ? "open" : "closed"]")
 
 
-/obj/item/gun/ballistic/automatic/l6_saw/afterattack(atom/target as mob|obj|turf, mob/living/user as mob|obj, flag, params) //what I tried to do here is just add a check to see if the cover is open or not and add an icon_state change because I can't figure out how c-20rs do it with overlays
+/obj/item/gun/ballistic/automatic/l6_saw/afterattack(atom/target as mob|obj|turf, mob/living/user as mob|obj, flag, params)
 	if(cover_open)
 		to_chat(user, "<span class='warning'>[src]'s cover is open! Close it before firing!</span>")
+		return
 	else
 		. = ..()
 		update_icon()
 
 //ATTACK HAND IGNORING PARENT RETURN VALUE
 /obj/item/gun/ballistic/automatic/l6_saw/attack_hand(mob/user)
-	if(loc != user)
-		..()
-		return	//let them pick it up
-	if(!cover_open || (cover_open && !magazine))
+	if (loc != user)
 		..()
-	else if(cover_open && magazine)
-		//drop the mag
-		magazine.update_icon()
-		magazine.forceMove(drop_location())
-		user.put_in_hands(magazine)
-		magazine = null
-		update_icon()
-		to_chat(user, "<span class='notice'>You remove the magazine from [src].</span>")
-		playsound(user, 'sound/weapons/magout.ogg', 60, 1)
+		return
+	if (!cover_open)
+		to_chat("<span class='warning'>[src]'s cover is closed! Open it before trying to remove the magazine!</span>")
```

Comment:
You forgot to include the mob argument, which would be user in this case. This message fails to show completely.

Reply:
FUCK